### ISSUE TITLE
Memory Gate Collapse: Optimizer Mismatch and Initialization Fixes

### ISSUE THREAD

**Issue by toddwbucy** [toddwbucy/Atlas-MAG_OmegaRule#20]
## Summary

Training run analysis revealed memory gates collapsed to ~0.002 (effectively zero), meaning the model completely ignored the long-term memory module. This was a **rational decision by the model** given the misaligned incentive structure.

### 1. Optimizer Mismatch (Critical)
- **Problem**: AdamW used for ALL parameters including memory internals
- **Impact**: AdamW (first-order) treats memory's noisy gradient landscape as noise and optimizes it away
- **Evidence**: Atlas paper Section 5 explicitly requires Muon/Newton-Schulz for memory updates
- **Current State**: TTL already implements Newton-Schulz for memory, but AdamW fights it during backprop

### 2. Conservative Gate Initialization
- **Problem**: Gates initialized to sigmoid(-3.0) ≈ 0.047 (early layers) to sigmoid(-1.0) ≈ 0.269 (later layers)
- **Impact**: 5% open gate is a "whisper" competing with attention's "megaphone"
- **Recommendation**: Initialize at 0.5+ to force signal through memory

### 3. GateMonitor Not Connected
- **Problem**: `GateMonitor` class exists but never instantiated in training loop
- **Impact**: Step 100 baseline and step 500 variance checks never fire
- **Evidence**: Gate collapse went undetected until post-hoc analysis

### 4. Task-Architecture Mismatch (Future Work)
- 512-token context means attention can see everything
- Memory module is redundant for short contexts
- Need long-context tasks (delayed code completion, NIAH probes)

### Priority 1: Exclude Memory from AdamW
```python
# Separate parameter groups
memory_params = {id(p) for block in model.blocks for p in block.memory.parameters()}
other_params = [p for p in model.parameters() if id(p) not in memory_params]
optimizer = AdamW(other_params, lr=config.learning_rate, ...)
```
TTL (already using Newton-Schulz) becomes the sole update mechanism for memory.

### Priority 2: Aggressive Gate Initialization
```python
# skeleton.py - change from:
gate_start = -3.0  # sigmoid ≈ 0.047
gate_end = -1.0    # sigmoid ≈ 0.269

# To:
gate_start = 0.0   # sigmoid = 0.5
gate_end = 0.5     # sigmoid ≈ 0.62
```

### Priority 3: Connect GateMonitor
```python
from src.training.gate_monitor import GateMonitor
gate_monitor = GateMonitor()
# In training loop:
gate_monitor.check(torch.tensor(model.get_gate_values()), global_step)
```

## Validation Criteria

After fixes, expect:
- [ ] Gate values stabilize in 0.3-0.6 range (not collapse to 0)
- [ ] Gate variance increases by step 500 (GateMonitor check passes)
- [ ] Memory contributes meaningfully to predictions

## References

- Atlas paper Section 5: Muon optimizer requirement
- Committee technical review (internal document)
- `src/training/gate_monitor.py` - existing but unused
- `src/training/ttl_update.py` - Newton-Schulz already implemented
